File: runinator-command-center/src/core/domain/models/workflow-state/map-frame.ts
import type { JsonValue } from "../../json";

export interface MapChild {
  index: number;
  child_run_id: string;
}

/** `state.map` bookkeeping (parent or child run). */
export interface MapFrame {
  node_id: string;
  target: string;
  items?: JsonValue[];
  concurrency?: number;
  next_index?: number;
  in_flight?: MapChild[];
  results?: JsonValue[];
  done?: number;
  item?: JsonValue | null;
  index?: number;
}

File: runinator-models/src/workflow_frames/map_frame.rs
#[allow(unused_imports)]
use super::*;

/// `state.map` parent fan-out bookkeeping or child item binding.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MapFrame {
    pub node_id: String,
    pub target: String,
    #[serde(default)]
    pub items: Vec<Value>,
    #[serde(default = "default_concurrency")]
    pub concurrency: i64,
    #[serde(default)]
    pub next_index: i64,
    #[serde(default)]
    pub in_flight: Vec<MapChild>,
    #[serde(default)]
    pub results: Vec<Value>,
    #[serde(default)]
    pub done: i64,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub item: Option<Value>,
    #[serde(default)]
    pub index: i64,
}
